src/input.js

```diff
@@ -211,20 +184,11 @@ class MutationHandler {
 		const viewPos = new ViewPosition( mutation.node, firstChangeAt );
 		const modelPos = this.editing.mapper.toModelPosition( viewPos );
 
-		// Remove appropriate number of characters from the model text node.
-		if ( deletions > 0 ) {
-			const removeRange = ModelRange.createFromPositionAndShift( modelPos, deletions );
-			this._remove( removeRange, deletions );
-		}
-
-		// Insert appropriate characters basing on `mutation.text`.
-		const insertedText = mutation.newText.substr( firstChangeAt, insertions );
-		this._insert( modelPos, insertedText );
-
-		// If there was `viewSelection` and it got correctly mapped, collapse selection at found model position.
-		if ( modelSelectionPosition ) {
-			this.editing.model.selection.collapse( modelSelectionPosition );
-		}
+		this.editor.execute( 'input', {
+			text: mutation.newText.substr( firstChangeAt, insertions ),
+			range: deletions > 0 ? ModelRange.createFromPositionAndShift( modelPos, deletions ) : null,
```

Comment:
Please move the consts out of this object literal. Those variables can make the code more readable (and easier to debug) thanks to their names.